src/tools.rs

```diff
@@ -514,27 +543,39 @@ mod archive {
         Ok(None)
     }
 
-    fn extract_file(mut read: impl Read, file: &str, target: &Path) -> Result<File> {
-        let out = target.join(file);
+    fn extract_file(mut read: impl Read, file: &str, target_directory: &Path) -> Result<File> {
+        let out = target_directory.join(file);
 
         if let Some(parent) = out.parent() {
             fs::create_dir_all(parent).context("failed creating output directory")?;
         }
 
-        let mut out = File::create(target.join(file)).context("failed creating output file")?;
+        let mut out =
+            File::create(target_directory.join(file)).context("failed creating output file")?;
         io::copy(&mut read, &mut out)
             .context("failed copying over final output file from archive")?;
 
         Ok(out)
     }
 
     /// Set the executable flag for a file. Only has an effect on UNIX platforms.
-    fn set_file_permissions(file: &mut File, mode: u32) -> Result<()> {
+    fn set_file_permissions(
+        file: &mut File,
+        mode: u32,
+        file_path_hint: Option<impl AsRef<str>>,
```

Comment:
I would also go for `impl Display` instead. That should be a better fit. But again, I can also do this post merge.